### PR TITLE
Feat/authentication

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Added Login, Signup, Logout functionality

- Login Page with working functionality
- Signup Page with working functionality
- Added a sidebar component for add notes page, after user logged in
- Logout button conditional rendering on Nav Top, logout functionality 

### Files to check

- [ ] src/contexts/auth-context.js
- [ ] src/utilities/loginFunc.js
- [ ] src/utilities/signupFunc.js
- [ ] src/pages/Authentication/LoginPage.jsx
- [ ] src/pages/Authentication/SignupPage.jsx
- [ ] src/pages/NotesPage/NotesPage.jsx
- [ ] src/components/Footer/Footer.jsx
- [ ] src/components/NavTop/NavTop.jsx

**NOTE: Ignore other files, they're index, images, styles**

### What not to check

- Notes Features

auth

### CODE REVIEW

File: src/components/NavTop/NavTop.jsx [hunk 1]
```diff
@@ -13,27 +28,33 @@ const NavTop = () => {
...
-            </Link>
-            </li>
-            <li className="notes-nav-item">
-              <a href="" className="btn">
+              <Link to="/addnote" className="btn">
                 Add Note
-              </a>
+              </Link>
             </li>
+            {console.log(auth.isAuth)}
```
Comment (devdeadviz) [inline src/components/NavTop/NavTop.jsx]: Remove `console log` statements.

**Review by devdeadviz — APPROVED** [review]
The code is clean. the file structure is good and every function and component can be reused. Good work!

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
+      <div className="page-container">
+        <NavTop />
+        <div className="content-wrap">
+          <Routes>
+            <Route path="/" element={<HomePage />} />
+            <Route path="/login" element={<LoginPage />} />
+            <Route path="/signup" element={<SignupPage />} />
+            <Route path="/addnote" element={<NotesPage />} />
+            <Route path="/mockman" element={<Mockman />} />
+          </Routes>
```
Comment (maitrakhatri) [inline src/App.js:20]: add route for 404 page

File: src/pages/Authentication/LoginPage.jsx [hunk 3]
```diff
@@ -1,26 +1,72 @@
...
+  const navigate = useNavigate();
+
+  const testLogin = {
+    email: "adarshbalika@gmail.com",
+    password: "adarshBalika123",
+  };
+
+  const loginSubmitHandler = async (user) => {
+    const { encodedToken, foundUser } = await loginFunc(user);
+    console.log(encodedToken, foundUser);
```
Comment (maitrakhatri) [inline src/pages/Authentication/LoginPage.jsx]: remove console logs before pushing the code

File: src/pages/Authentication/SignupPage.jsx [hunk 4]
```diff
@@ -1,44 +1,152 @@
...
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+  const redirect = useNavigate();
+  const { setAuth } = useAuth();
+
+  const signupSubmitHandler = async (user) => {
+    const { encodedToken, createdUser } = await signupFunc(user);
+    console.log("from signupSubmitHandler data", encodedToken, createdUser);
```
Comment (maitrakhatri) [inline src/pages/Authentication/SignupPage.jsx]: same as above

File: src/pages/Authentication/signup.css [hunk 5]
```diff
@@ -4,13 +4,18 @@
   line-height: 1.5;
 }
 
+.form-container .pswrd-match {
+  color: red;
```
Comment (maitrakhatri) [inline src/pages/Authentication/signup.css]: use variables or hex code for colors

File: src/utilities/signupFunc.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+const signupFunc = async (user) => {
+  try {
+    const { data } = await axios.post("/api/auth/signup", {
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      password: user.password,
+    });
+    console.log("from signupFunc data", data );
```
Comment (maitrakhatri) [inline src/utilities/signupFunc.js]: same as above

**Review by maitrakhatri — APPROVED** [review]
Code and logic is very easy to understand
Really liked the folder structure and the fact that you have used variables for spacing. 
Have left some suggestions

**Pull request merged**